python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -605,6 +603,7 @@ def on_output_taken(self, output: RefBundle):
         """Callback when an output is taken from the operator."""
         self.num_outputs_taken += 1
         self.rows_task_outputs_generated += output.num_rows() or 0
+        self.num_task_outputs_generated += len(output)
```

Comment:
I just realized that this is not correct.
`num_task_outputs_generated` and `rows_task_outputs_generated` are supposed to be measure the "task outputs".
We need them in addition to `num_outputs_taken` and `bytes_outputs_taken`, because some ops don't launch tasks that generate outputs. They just pass through the input blocks to the output. E.g., streaming_split and union. 
We should revert this change. and add the missing `rows_outputs_taken` metric.